api/envoy/config/filter/http/ext_authz/v2alpha/ext_authz.proto

```diff
@@ -42,24 +42,29 @@ message ExtAuthz {
 // whether the request is authorized or not.
 //
 // A successful check allows the authorization service adding or overriding headers from the
-// original request before dispatching it to the upstream. This is done by including the headers in
-// the response sent back from the authorization service to the filter. Note that `Status`,
-// `Method`, `Path` and `Content Length` response headers are automatically removed from this
-// response by the filter. If other headers need be deleted, they should be specified in
-// `response_headers_to_remove` field.
+// original request before dispatching it to the upstream. This is done by configuring which headers
+// in the authorization response should be sent to the upstream. See *allowed_authorization_headers*
+// bellow.
 //
 // A failed check will cause this filter to close the HTTP request normally with 403 (Forbidden),
-// unless a different status code has been indicated by the authorization service via response
-// headers. The HTTP service also allows the authorization filter to also pass data from the
-// response body to the downstream client in case of a denied request.
+// unless a different status code has been indicated by the authorization server via response
+// headers. If other headers in the authorization response need to be send to client, this can also
+// be done by specifying them in *allowed_authorization_headers*.
 message HttpService {
   // Sets the HTTP server URI which the authorization requests must be sent to.
   envoy.api.v2.core.HttpUri server_uri = 1;
 
   // Sets an optional prefix to the value of authorization request header `path`.
   string path_prefix = 2;
 
-  // Sets a list of headers that should be not be sent *from the authorization server* to the
-  // upstream.
-  repeated string response_headers_to_remove = 3;
+  // Sets a list of headers that can be sent from the authorization server to the upstream or to the
+  // downstream when present in the authorization response. Note that a matched request header will
+  // have its value overriden by the ones sent from the authorization server.
+  repeated string allowed_authorization_headers = 3;
```

Comment:
I'm not sure of the production status of this filter? Can we just change this? Or do we need to deprecate and use new fields? (IMO that's probably safer. I don't know that we need to fully deprecate but you could reserve 3 and allocate 4 and 5?)